sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchClient.java

```diff
@@ -551,7 +623,25 @@ public <T> T getDocument(String key, Class<T> modelClass) {
     @ServiceMethod(returns = ReturnType.SINGLE)
     public <T> Response<T> getDocumentWithResponse(String key, Class<T> modelClass, List<String> selectedFields,
         Context context) {
-        return asyncClient.getDocumentWithResponse(key, modelClass, selectedFields, context).block();
+
+        try {
+            Response<Object> response = restClient.getDocuments()
+                .getWithResponse(key, selectedFields, null, Utility.enableSyncRestProxy(context));
+            if (serializer == null) {
+                return new SimpleResponse<>(response, Utility.convertValue(response.getValue(), modelClass));
+            }
+            ByteArrayOutputStream sourceStream = new ByteArrayOutputStream();
+            serializer.serialize(sourceStream, response.getValue());
+            T doc = serializer.deserialize(new ByteArrayInputStream(sourceStream.toByteArray()),
+                createInstance(modelClass));
+            return new SimpleResponse<>(response, doc);
+        } catch (SearchErrorException ex) {
+            throw LOGGER.logExceptionAsError(DocumentResponseConversions.mapSearchErrorException(ex));
```

Comment:
@alzimmermsft, should we log this exception here ?
is this already logged somewhere in core pipeline flow ?

Reply:
It's likely already logged somewhere else in the pipeline but I'm not 100% certain where this would happen. I think we can punt on this for a holistic discussion on logging and prevention of duplicative error logging.